### PR TITLE
release.yml: Release tarball with GitHub Actions

### LINKED ISSUE #5: Cross-compile vde_vmnet for ARM target
```console
$ CFLAGS="-target arm64-macos" LDFLAGS="-target arm64-macos" make
cc -target arm64-macos -DVERSION=\"v0.1.0\" -c cli.c -o cli.o
cc -target arm64-macos -DVERSION=\"v0.1.0\" -c main.c -o main.o
cc -target arm64-macos -DVERSION=\"v0.1.0\" -o vde_vmnet -target arm64-macos -lvdeplug -framework vmnet  cli.o  main.o
ld: warning: ignoring file /usr/local/lib/libvdeplug.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
Undefined symbols for architecture arm64:
  "_vde_close", referenced from:
      _main in main.o
  "_vde_open_real", referenced from:
      _main in main.o
  "_vde_recv", referenced from:
      _main in main.o
  "_vde_send", referenced from:
      __on_vmnet_packets_available in main.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [vde_vmnet] Error 1
```

### LINKED ISSUE #22: Include a tarball with pre-compiled binaries in vde_vmnet releases
The tarball should include both `vde_switch` and `vde_vmnet` binaries with an `/opt/vde` prefix compiled in.

### PR DESCRIPTION
**Pull request by reishoku**
This PR contains:
- YAML file for GitHub Actions to automatically release precompiled FAT binary (supports both x86_64 and arm64) when commits with specific tag name were pushed to the repository.

closes: https://github.com/lima-vm/vde_vmnet/issues/5
closes: https://github.com/lima-vm/vde_vmnet/issues/22

----------------------------------------
Signed-off-by: KOSHIKAWA Kenichi <reishoku.misc ~~at~~ pm.me>

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+on:
+  push:
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+
+jobs:
+  build:
+    name: build
+    strategy:
+      matrix:
+        platform: [macos-10.15, macos-11]
```
Comment (AkihiroSuda) [inline .github/workflows/release.yml]: Probably we don't need matrix

Comment (reishoku) [reply .github/workflows/release.yml]: Thanks for the comment. Fixed.

**Review by jandubois — DISMISSED** [review]
Thank you, LGTM. I haven't actually tested the FAT binaries.

I'm not sure though what benefit we get from using `retention-days: 1`. Doesn't that mean that even on a failed action, the logs will be deleted after one day? So I would have either left it at the default, or set it to at least 5 or 10. Or is this only for artifacts, but not for logs?

**Comment by reishoku** [comment]
Thanks for the comment. I found some mistakes, so I made some changes to the PR.

> is this only for artifacts, but not for logs?

As for this, both artifacts and logs will be deleted, it seems.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+          export CC=$(xcrun --sdk macosx --find clang)
+          export CXX=$(xcrun --sdk macosx --find clang++)
+          export CFLAGS="-arch x86_64 -arch arm64e -isysroot $SDKROOT -Wno-error=implicit-function-declaration"
+          export CXXFLAGS=$CFLAGS
+          ./configure --prefix=/opt/vde
+          make PREFIX=/opt/vde
+          sudo make PREFIX=/opt/vde install
+      - name: Build and Make vde_vmnet
+        run: |
+          git clone https://github.com/lima-vm/vde_vmnet.git /tmp/vde_vmnet
```
Comment (AkihiroSuda) [inline .github/workflows/release.yml:39]: Use `actions/checkout`

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+          cd /tmp/vde_vmnet
+          make PREFIX=/opt/vde
+          sudo make PREFIX=/opt/vde install
```
Comment (AkihiroSuda) [inline .github/workflows/release.yml:42]: Seems only built for amd64

**Comment by AkihiroSuda** [comment]
Let me close this, but thanks anyway for contribution

**Pull request closed without merging**